### PR TITLE
feat: add `fmt_icon()` method

### PR DESCRIPTION
**Pull request by rich-iannone**
This PR adds the `fmt_icon()` method. It depends on Font Awesome icons available in the `faicons` library, so that is a new dependency. Whenever that package is updated, new icons from the FA free set will be available. Here's an example of the `fmt_icon()` method in use:

```python
import pandas as pd
from great_tables import GT
animals_foods_df = pd.DataFrame(
    {
        "animals": ["hippo", "fish,spider", "mosquito,locust,frog", "dog,cat", "kiwi-bird"],
        "foods": ["bowl-rice", "egg,pizza-slice", "burger,lemon,cheese", "carrot,hotdog", "bacon"],
    }
)
(
    GT(animals_foods_df)
    .fmt_icon(
        columns=["animals", "foods"],
        height="4em"
    )
    .cols_align(
        align="center",
        columns=["animals", "foods"]
    )
)
```

image

### CODE REVIEW

File: great_tables/_formats.py [hunk 1]
```diff
@@ -3830,6 +3832,289 @@ def _build_img_tag(uri: str, height: str | None = None, width: str | None = None
...
+            height = "1em"
+        else:
+            height = self.height
+
+        if self.stroke_width is None:
+            stroke_width = "1px"
+        elif isinstance(self.stroke_width, (int, float)):
+            stroke_width = f"{str(self.stroke_width)}px"
+        else:
+            stroke_width = self.stroke_width
```
Comment (machow) [inline great_tables/_formats.py:4074]: Please test this

File: great_tables/_formats.py [hunk 2]
```diff
@@ -3830,6 +3832,289 @@ def _build_img_tag(uri: str, height: str | None = None, width: str | None = None
...
+        out: list[str] = []
+
+        for icon in icon_list:
+
+            if isinstance(self.fill_color, dict):
+
+                if icon in self.fill_color:
+                    fill_color = self.fill_color[icon]
+                else:
+                    fill_color = None
```
Comment (machow) [inline great_tables/_formats.py:4085]: Please test this

File: great_tables/_formats.py [hunk 3]
```diff
@@ -3830,6 +3832,289 @@ def _build_img_tag(uri: str, height: str | None = None, width: str | None = None
...
+            pl.col("population_2021").sort(descending=True).head(5)
+        ])
+        .sort("csd_type")
+        .explode(["name", "population_2021"])
+        .with_columns(
+            csd_type = pl.when(pl.col("csd_type") == "town")
+            .then(pl.lit("house-chimney"))
+            .otherwise(pl.lit("city"))
+        )
+    )
```
Comment (machow) [inline great_tables/_formats.py:3967]: Is it possible to just select a few rows of this data. E.g. `towny[[0, 3, 5], :]`. Say maybe the top 2 in each of your groups. It seems easier to get into an example with less setup (a doesn't seem critical to have a comprehensive table).

**Review by machow — CHANGES_REQUESTED** [review]
This LGTM, if we can polars data wrangling in the second example

File: great_tables/_formats.py (hunk 1 above)
Comment (rich-iannone) [reply great_tables/_formats.py:4042]: Now tested.

File: great_tables/_formats.py (hunk 2 above)
Comment (rich-iannone) [reply great_tables/_formats.py:4051]: Tests now present for this.

**Review by machow — APPROVED** [review]
LGTM, thanks!

**Pull request merged**